dvc/state.py

```diff
@@ -29,11 +29,18 @@ def __init__(self, dvc_version, expected, actual):
         )
 
 
-def _file_metadata_changed(actual_mtime, mtime, actual_size, size):
-    return actual_mtime != mtime or actual_size != size
+class StateBase(object):
```

Comment:
What's the reason behind having another Base/Abstract class? The only one who is using `StateBase` is `State`, so it doesn't make a lot of sense to me.

It doesn't even raise `NotImplementedError`, just passes silently.